src/mtev_http.c

```diff
@@ -317,6 +328,35 @@ struct bchain *bchain_from_data(const void *d, size_t size) {
   return n;
 }
 
+static mtev_http_compress 
+request_compression_type(mtev_http_request *req)
+{
+  const char *content_encoding = NULL;
+  if (req == NULL || req->freed) return MTEV_HTTP_COMPRESS_NONE;
+  
+  mtev_hash_table *headers = mtev_http_request_headers_table(req);
+  mtev_hash_retr_str(headers, "content-encoding", strlen("content-encoding"), 
+                     &content_encoding);
+
+  if (content_encoding == NULL) {
+    return MTEV_HTTP_COMPRESS_NONE;
+  }
+  
+  /* there is no official mime-type for LZ4 so use this for now */
+  int lz4f = strcmp("lz4f", content_encoding);
```

Comment:
recommend adding `#define` macros to the .c file:
```
#define CONTENT_ENCODING_GZIP "gzip"
#define CONTENT_ENCODING_COMPRESS "compress"
/* no official mime-type for LZ4 so use this for now */
#define CONTENT_ENCODING_LZ4F "x-lz4f"
```
so that that implementation caveat is documented, and reading any part of the code that refers to LZ4F encoding naturally sends them near that comment.